components/com_content/src/Model/CategoryFormModel.php

```diff
@@ -0,0 +1,501 @@
+                $parentId = (int) ($item->parent_id ?? 0);
+                $canEdit  = $user->authorise('core.create', $asset)
+                    || $user->authorise('core.create', 'com_content.category.' . $parentId);
+            } elseif ($user->authorise('core.edit', $asset)) {
+                $canEdit = true;
+            } elseif (!empty($userId) && $user->authorise('core.edit.own', $asset)) {
+                $canEdit = (int) $userId === (int) $item->created_user_id;
+            }
+        }
+
+        $item->accessEdit   = $canEdit;
+        $item->accessChange = $frontendEdit !== false && $user->authorise('core.edit.state', $asset);
+
+        if ($item->id) {
+            $tagsHelper = new TagsHelper();
+            $item->tags = $tagsHelper->getTagIds($item->id, $item->extension . '.category');
+        }
+
+        return $item;
+    }
+
+    /**
+     * Get the return URL.
+     *
+     * @return  string  The return URL.
+     *
+     * @since   6.1.0
+     */
+    public function getReturnPage()
+    {
+        return base64_encode($this->getState('return_page', ''));
+    }
+
+    /**
+     * Method to get the data that should be injected in the form.
+     *
+     * @return  mixed  The data for the form.
+     *
+     * @since   6.1.0
+     */
+    protected function loadFormData()
+    {
+        $app  = Factory::getApplication();
+        $data = $app->getUserState('com_content.edit.category.data', []);
+        $stateId = (int) $this->getState('category.id');
+
+        if (empty($data)) {
+            $data = $this->getItem();
+
+            if (empty($data->id)) {
+                $data->published = $app->getInput()->getInt('published', $data->published ?? null);
+                $data->language  = $app->getInput()->getCmd('language', $data->language ?? null);
+                $data->access    = $app->getInput()->getInt('access', $app->get('access'));
+            }
+        } else {
+            // Drop stale session data if it belongs to another category.
+            if (isset($data['id']) && (int) $data['id'] !== $stateId) {
+                $app->setUserState('com_content.edit.category.data', null);
+                $data = $this->getItem();
+            }
+
+            $data = (object) $data;
+        }
+
+        // Use the core categories context so plugins process category data as expected.
+        $this->preprocessData('com_categories.category', $data);
+
+        // Ensure tags are stored as a simple array.
+        if (isset($data->tags) && $data->tags instanceof TagsHelper) {
+            $data->tags = $data->tags->getTagIds($data->id ?? 0, $this->option . '.category');
+        }
+
+        return $data;
+    }
+
+    /**
+     * Method to get the row form.
+     *
+     * @param   array    $data      Data for the form.
+     * @param   boolean  $loadData  True if the form is to load its own data (default case), false if not.
+     *
+     * @return  Form  A Form object
+     *
+     * @since   6.1.0
```

Comment:
```suggestion
     * @since   __DEPLOY_VERSION__
```

doing this means our build scripts will correctly set the version - it might not make it to 6.1 ;)